language-plutus-core/budgeting-bench/Bench.hs

```diff
@@ -1,5 +1,158 @@
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE MagicHash           #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeOperators       #-}
+
 module Main (main) where
 
+import           Language.PlutusCore
+-- import           Language.PlutusCore.Constant.Dynamic
+import           Language.PlutusCore.Evaluation.Machine.Cek
+-- import           Language.PlutusCore.Evaluation.Machine.ExBudgeting
+import           Language.PlutusCore.Evaluation.Machine.ExBudgetingDefaults
+import           Language.PlutusCore.Evaluation.Machine.ExMemory
+-- import           Language.PlutusCore.Evaluation.Machine.ExMemory
+-- import           Language.PlutusCore.FsTree
+-- import           Language.PlutusCore.Generators.Interesting
+import qualified Data.ByteString.Lazy                                       as BSL
+import           Hedgehog
+import           Hedgehog.Internal.Gen
+import           Hedgehog.Internal.Tree
+import           Hedgehog.Range
+import           Language.PlutusCore.MkPlc
+
+-- import           Control.Lens
+import           Criterion.Main
+import qualified Criterion.Types                                            as C
+
+runTermBench :: String -> Plain Term DefaultUni -> Benchmark
+runTermBench name term = env
+    (do
+        (_result, budget) <-
+          pure $ runCekCounting mempty defaultCostModel term
+        -- print result
+        -- print (budget ^. (exBudgetStateBudget.exBudgetCPU) :: ExCPU)
+        pure budget
+        )
+    (\_ -> bench name $ nf (unsafeEvaluateCek mempty defaultCostModel) term)
+
+-- bunchOfFibs :: PlcFolderContents DefaultUni
+-- bunchOfFibs =
+--     let
+--         fibFile i = plcTermFile (show i) (naiveFib i)
+--     in
+        -- FolderContents [ treeFolderContents "Fib" (fibFile <$> [1..10]) ]
+
+benchSameTwoByteStrings :: BuiltinName -> Benchmark
+benchSameTwoByteStrings name = createTwoTermBuiltinBench name bsA bsA
+    where
+        bsA = (expToBenchingBytestring seedA . fromInteger) <$> expsToBench
+
+benchTwoByteStrings :: BuiltinName -> Benchmark
+benchTwoByteStrings name = createTwoTermBuiltinBench name bsA bsB
+    where
+        bsA = (expToBenchingBytestring seedA . fromInteger) <$> expsToBench
+        bsB = (expToBenchingBytestring seedB . fromInteger) <$> expsToBench
+
+benchBytestringOperations :: BuiltinName -> Benchmark -- TODO the numbers are a bit too big here
+benchBytestringOperations name = createTwoTermBuiltinBench @Integer @BSL.ByteString name numbers bsA
+    where
+        bsA = (expToBenchingBytestring seedA . fromInteger) <$> expsToBench
+        numbers = expToBenchingInteger <$> expsToBench
+
+createTwoTermBuiltinBench :: (DefaultUni `Includes` a, DefaultUni `Includes` b) => BuiltinName -> [(a, ExMemory)] -> [(b, ExMemory)] -> Benchmark
+createTwoTermBuiltinBench name as bs =
+    bgroup (show name) $
+        (flip fmap) as (\(x, xMem) ->
```

Comment:
I think `flip fmap as f` is better spelled as `as <&> f`.

And in other places.